server/src/main/scala/replpp/server/WebServiceWithWebSocket.scala

```diff
@@ -0,0 +1,96 @@
+package replpp.server
+
+import cask.model.Response.Raw
+import cask.model.{Request, Response}
+import cask.router.Result
+import org.slf4j.{Logger, LoggerFactory}
+import ujson.Obj
+
+import java.util.concurrent.ConcurrentHashMap
+import java.util.{Base64, UUID}
+import scala.util.{Failure, Success, Try}
+
+trait HasUUID { def uuid: UUID }
+
+case class UsernamePasswordAuth(username: String, password: String)
+
+abstract class WebServiceWithWebSocket[T <: HasUUID](
+  override val host: String,
+  override val port: Int,
+  authenticationMaybe: Option[UsernamePasswordAuth] = None) extends cask.MainRoutes {
+  protected val logger: Logger = LoggerFactory.getLogger(getClass)
+
+  class basicAuth extends cask.RawDecorator {
+    def wrapFunction(request: Request, delegate: Delegate): Result[Raw] = {
+      val isAuthorized = authenticationMaybe match {
+        case None => true // no authorization required
+        case Some(requiredAuth) =>
+          parseAuthentication(request) match {
+            case None => false // no authentication provided
+            case Some(providedAuth) =>  providedAuth == requiredAuth
```

Comment:
makes sense and that actually simplifies the code. 
I changed it to `MessageDigest.isEqual(providedAuth.toString.getBytes(utf8), expectedAuth.toString.getBytes(utf8))` since we also want to compare the username, not only the password.